scripts/resolve_container_path.py

```diff
@@ -19,24 +27,44 @@
 
 def resolve_container_path(container_path):
     container_path = os.path.realpath(container_path)
+    mountinfo_path = '/proc/self/mountinfo'
     try:
-        with open('/proc/self/cgroup') as f:
-            # Entries in /proc/self/cgroup look like this (note the nesting):
-            # 11:name=systemd:/docker/82c1bd2…23b5bcf/docker/6547bce…60ca5a7
-            prefix, container_id = next(f).strip().split(':')[2].split('/')[-2:]
+        with open(mountinfo_path) as f:
+            log.info('Found %s', mountinfo_path)
+            # Entries of interest in /proc/self/mountinfo look as follows:
+            # 752 744 259:2 /docker/containers/dc61d9…/…
+            # dc61d9… is the container ID
+            for line in f:
+                assert line.endswith('\n'), line
+                parts = line.strip().split(' ')
+                assert len(parts) > 9, parts
+                root = parts[3]
+                for prefix in ('/var/lib/docker/containers/', '/docker/containers/'):
+                    if root.startswith(prefix):
+                        log.info('Extracting container ID from %s', root)
+                        container_id = root[len(prefix):].split('/')[0]
```

Comment:
A previous version employed a set to ensure that all container IDs are the same. Why did you remove the set? This code is now silently accepting multiple matching lines with inconsistent container IDs, arbitrarily using the last one found.